nestkernel/spatial.cpp

```diff
@@ -447,11 +447,7 @@ minus_mask( const MaskDatum& mask1, const MaskDatum& mask2 )
 void
 connect_layers( NodeCollectionPTR source_nc, NodeCollectionPTR target_nc, const DictionaryDatum& connection_dict )
 {
-  const thread num_threads = kernel().vp_manager.get_num_threads();
-  for ( thread tid = 0; tid < num_threads; ++tid )
-  {
-    kernel().connection_manager.set_have_connections_changed( tid );
-  }
+  kernel().connection_manager.set_have_connections_changed( kernel().vp_manager.get_thread_id() );
```

Comment:
I'd move this to right before `source->connect()` with a similar comment as suggested above.